File: Chapter 03/C#/Section_3_3_3_LSP/Before/Player.cs
namespace Chapter03.LSP.Before;

/// <summary>
/// The Base Contract.
/// 
/// ARCHITECTURE NOTE: By placing this method in the base abstract class, we are 
/// creating a strict contract: "Every single class that inherits from Player 
/// MUST be able to execute PlayFieldPosition() successfully."
/// </summary>
public abstract class Player
{
    public abstract void PlayFieldPosition();
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_3_lsp/before/Player.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_3_lsp.before;

/**
 * The Base Contract.
 * * ARCHITECTURE NOTE: By placing this method in the base abstract class, we are 
 * creating a strict contract: "Every single class that inherits from Player 
 * MUST be able to execute playFieldPosition() successfully."
 */
public abstract class Player {
    public abstract void playFieldPosition();
}